app/assets/javascripts/course/discussion/topics.js

```diff
@@ -0,0 +1,37 @@
+(function($) {
+  'use strict';
+  var DOCUMENT_SELECTOR = '.course-discussion-topics.index ';
+
+  // The comment boxes are hidden by default, show the boxes if the browser supports javascript.
+  function showCommentBoxes() {
+    var $forms = $('.post-form').filter(DOCUMENT_SELECTOR + '*');
+    $forms.show();
+  }
+
+  function onPostFormSubmit(e) {
+    e.preventDefault();
+
+    var $form = $(e.target);
+    var action = $form.attr('action');
+    var method = $form.attr('method');
+
+    $.ajax({ url: action, method: method, data: $form.serialize() }).
```

Comment:
There are several places where forms are submitted via AJAX, and then disabling the actual HTML form submission. This whole behaviour should be extracted out